packages/task/src/node/task-server.ts

```diff
@@ -0,0 +1,258 @@
+                        taskId: task.id,
+                        terminalId: (task.processType === 'terminal') ? task.process.id : undefined,
+                        processId: (task.processType === 'raw') ? task.process.id : undefined,
+                        osProcessId: task.process.pid,
+                        command: task.command,
+                        label: task.label
+                    });
+            }
+        }
+        return Promise.resolve(taskinfo);
+    }
+
+    run(task: TaskOptions, ctx?: string): Promise<TaskInfo> {
+        return this.doRun(task, ctx);
+    }
+
+    protected doRun(options: TaskOptions, ctx?: string): Promise<TaskInfo> {
+        return new Promise<TaskInfo>(async (resolve, reject) => {
+            // on windows, prefer windows-specific options, if available
+            const processOptions = (isWindows && options.windowsProcessOptions !== undefined) ?
+                options.windowsProcessOptions : options.processOptions;
+
+            const command = processOptions.command;
+            const cwd = FileUri.fsPath(options.cwd);
+
+            // Use task's cwd with spawned process and pass node env object to
+            // new process, so e.g. we can re-use the system path
+            processOptions.options = {
+                cwd: cwd,
+                env: process.env
+            };
+
+            // When we create a process to execute a command, it's difficult to know if it failed
+            // because the executable or script was not found, or if it was found, ran, and exited
+            // unsuccessfully. So here we look to see if it seems we can find a file of that name that
+            // is likely to be the one we want, before attemting to execute it.
+            this.findCommand(command, cwd)
+                .then(cmd => {
+                    try {
+                        // use terminal or raw process
+                        let task: Task;
+                        let proc: TerminalProcess | RawProcess;
+
+                        if (options.processType === 'terminal') {
+                            this.logger.info('Task: creating underlying terminal process');
+                            proc = this.terminalProcessFactory(<TerminalProcessOptions>processOptions);
+                        } else {
+                            this.logger.info('Task: creating underlying raw process');
+                            proc = this.rawProcessFactory(<RawProcessOptions>processOptions);
+                        }
+
+                        task = this.taskFactory(<TaskProcessOptions>
+                            {
+                                label: options.label,
+                                command: cmd,
+                                process: proc,
+                                processType: options.processType,
+                                context: ctx
+                            });
+
+                        const toDispose = new DisposableCollection();
```

Comment:
seems like there is always only one disposable in the collection so we could remove it.

Reply:
done